infrastructure/src/main/java/org/corfudb/infrastructure/log/FileSystemAgent.java

```diff
@@ -68,7 +72,7 @@ public static ResourceQuota getResourceQuota() {
         return instance.orElseThrow(err).logSizeQuota;
     }
 
-    public static PartitionAttribute getPartition() {
+    public static PartitionAttribute getPartitionAttribute() {
         Supplier<IllegalStateException> err = () -> new IllegalStateException(NOT_CONFIGURED_ERR_MSG);
         return instance.orElseThrow(err).partitionAttribute;
```

Comment:
This can return null since the `FileSystemAgent` constructor can call `getPartitionAttribute()` before `PartitionAgent` has a chance to set the value in the other thread. This can cause an NPE in `RemoteMonitoringService`.

Reply:
fixed